crates/primitives/src/utils.rs

```diff
@@ -52,3 +50,105 @@ pub fn get_address(buf: &[u8]) -> Option<Address> {
         None
     }
 }
+
+pub fn unpack_paymaster_fee_data(buf: &[u8]) -> (Address, U256, U256, Bytes) {
```

Comment:
I think we can rename this to `unpack_paymaster_data`, and `unpack_factory_data` ...